### PR TITLE
fix: changed decrypt passwords to false

### PR DESCRIPTION
**Pull request by GregoryIvo**
> **Note**: We are using the Conventional Commits convention for our pull request titles. Please take a look at the [PR title format document](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md#submitting-the-changes) for the supported [types](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md#type) and [scopes](https://github.com/eclipse/kura/blob/develop/CONTRIBUTING.md#scope).

Brief description of the PR.

Fix a configurableComponents/configurations/byPid REST API error, where attempting to decode a password throws an error.

**Related Issue:** N/A

**Description of the solution adopted:** N/A

**Screenshots:** N/A

**Any side note on the changes made:** N/A

### CODE REVIEW

**Comment by GregoryIvo** [comment]
Still working on the tests for this PR

**Comment by GregoryIvo** [comment]
PR is now ready for review!

**Comment by MMaiero** [comment]
I think we need to add tests to prevent regressions in the /byPid call.
I may have missed, but I think that tests now are not covering a password field in the /byPid or /byPid/_default, right?

**Review by MMaiero — CHANGES_REQUESTED** [review]
I think we need to add tests to prevent regressions in the /byPid call.
I may have missed, but I think that tests now are not covering a password field in the /byPid or /byPid/_default, right?

**Comment by GregoryIvo** [comment]
> I think we need to add tests to prevent regressions in the /byPid call.
> 
> I may have missed, but I think that tests now are not covering a password field in the /byPid or /byPid/_default, right?

Seems to be so.

**Comment by GregoryIvo** [comment]
After going over the tests, we may be good.  Let me know what you think?

image

File: kura/org.eclipse.kura.rest.configuration.provider/src/main/java/org/eclipse/kura/internal/rest/configuration/ConfigurationRestService.java [hunk 1]
```diff
@@ -148,15 +150,20 @@ public PidSet listFactoryComponentsPids() {
...
+     * The {@link FactoryComponentConfigurationDTO} will provide all the information
+     * needed to generate the instance: it
+     * links the factory Pid to be used, the target instance Pid, the properties to
+     * be used when creating the instance
      * and if the request should be persisted with a snapshot.
      * In case of a request error, an exception is thrown.
      *
      * @param factoryComponentConfiguration
-     *            provides all the parameters needed to generate a new instance from a Factory Component
+     *                                      provides all the parameters needed to
```
Comment (MMaiero) [inline kura/org.eclipse.kura.rest.configuration.provider/src/main/java/org/eclipse/kura/internal/rest/configuration/ConfigurationRestService.java:164]: Can we double check the formatting here? Spacing seems odd. Are we using the Kura Formatter and cleanup profiles?

File: kura/test/org.eclipse.kura.core.cloud.test/pom.xml [hunk 2]
```diff
@@ -32,41 +31,41 @@
 		<sonar.coverage.jacoco.xmlReportPaths>${project.build.directory}/site/jacoco-aggregate/jacoco.xml</sonar.coverage.jacoco.xmlReportPaths>
 	</properties>
 
-    <build>
-    	<plugins>
+	<build>
```
Comment (MMaiero) [inline kura/test/org.eclipse.kura.core.cloud.test/pom.xml]: same here.

**Comment by MMaiero** [comment]
> After going over the tests, we may be good. Let me know what you think?
> 
> image

Yes, I see coverage. But do the test(s) that stresses those areas have a password field to verify the actual behaviour?

**Comment by GregoryIvo** [comment]
1) attempted to reformat the suggested docs. They seem to be already formatted correctly. (nothing changed on my end, and pasting my current config settings for sanity check)
image

2) will make the tests.

**Comment by MMaiero** [comment]
> 1. attempted to reformat the suggested docs. They seem to be already formatted correctly. (nothing changed on my end, and pasting my current config settings for sanity check)
> 
> image
> 
> 2. will make the tests.

Rethinking of it I believe the formatter covers just java code not xml. It may be that the xml is configured with tabs instead of spaces in your setup?
Maybe that can explain the change in the PR.

Thanks for 2).

**Comment by MMaiero** [comment]
@GregoryIvo Can you please rebase so that we solve the conflict with the changes made in a previous PR? Thanks

**Comment by GregoryIvo** [comment]
@MMaiero I have finally made the suggested tests, let me know what you think!

**Review by MMaiero — APPROVED** [review]

**Pull request merged**